api/src/test/java/org/openmrs/api/VisitServiceTest.java

```diff
@@ -938,21 +937,25 @@ public void saveVisit_shouldAssociateEncounterWithVisitOnSaveEncounter() {
 	/**
 	 * @see VisitService#saveVisit(Visit)
 	 */
-	@Test(expected = TransientObjectException.class)
+	@Test
```

Comment:
This was a bad test, hibernate throws this exception for the other side of the relationship i.e. if you save an encounter where visit is transient and not the opposite

Reply:
If this is a bug fix, could we separate it from the switch to annotations?

Reply:
Again, this is masked by xml config, to reproduce it, the annotation changes must exist in the codebase already which will break CI (chicken-egg dilemma) which is why I have just fixed them in one PR